### PR TITLE
rubocop

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/samson 

some todos left over ... but let's get the base merged first :)

### CODE REVIEW

**Comment by jonmoter** [comment]
I'm not reviewing a PR with 282 files that changed and the CI builds are failing.

**Comment by grosser** [comment]
all fixed now ... let me know if you want chunked PRs instead to reduce the size/scope ...

**Comment by sandlerr** [comment]
can we turn off `Style/IfUnlessModifier`?

File: config/initializers/app_routes.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@ module AppRoutes
   # memoize this due to a bug in MRI that shows up in multi-threaded code
   # see https://github.com/puma/puma/issues/647, search for "bug in MRI"
   def self.url_helpers
-    @@routes_url_helpers ||= Rails.application.routes.url_helpers
+    @routes_url_helpers ||= Rails.application.routes.url_helpers
```
Comment (sandlerr) [inline config/initializers/app_routes.rb]: any behaviour change here? looks to me like we can delete the entire file now?

File: plugins/env/app/models/environment_variable.rb [hunk 2]
```diff
@@ -40,7 +41,7 @@ def matches?(ev, deploy_group)
     def resolve_dollar_variables(env)
       env.each_value do |value|
         value.gsub!(/\$\{(\w+)\}|\$(\w+)/) do |original|
-          env[$1 || $2] || original
+          env[Regexp.last_match(1) || Regexp.last_match(2)] || original
```
Comment (sandlerr) [inline plugins/env/app/models/environment_variable.rb]: not sure this is an improvement

File: config/initializers/app_routes.rb (hunk 1 above)
Comment (grosser) [inline config/initializers/app_routes.rb]: it is a class method, so `@` should be fine / a global cache, why would we want to delete the file ?

File: plugins/env/app/models/environment_variable.rb (hunk 2 above)
Comment (grosser) [inline plugins/env/app/models/environment_variable.rb]: yeah true ... did not see that ...

**Comment by grosser** [comment]
do you have an example for `Style/IfUnlessModifier` ? / why you don't like it ?

**Comment by grosser** [comment]
@sandlerr broke out the app changes here https://github.com/zendesk/samson/pull/1004 ... let's start with that :)

File: config/initializers/app_routes.rb (hunk 1 above)
Comment (sandlerr) [inline config/initializers/app_routes.rb]: the Ruby bug seems to have been closed in 2.2

**Comment by grosser** [comment]
PR 1000 is ready now @sandlerr ... but are you ready for PR 1000 ?

File: plugins/datadog/app/models/datadog_notification.rb [hunk 3]
```diff
@@ -23,7 +24,7 @@ def deliver
     )
 
     client = Dogapi::Client.new(api_key, nil, "")
-    status, _ = client.emit_event(event)
+    status, = client.emit_event(event)
```
Comment (sandlerr) [inline plugins/datadog/app/models/datadog_notification.rb]: this is weird syntax, is it equivalent to `status = client.emit_event(event)[0]` ?

**Comment by sandlerr** [comment]
this has merge conflicts & codeclimate errors now :/

File: plugins/datadog/app/models/datadog_notification.rb (hunk 3 above)
Comment (grosser) [inline plugins/datadog/app/models/datadog_notification.rb]: yeah ... theoretically correct, but looks fishy :D

**Comment by grosser** [comment]
rebased and fixed syntax

**Comment by grosser** [comment]
codeclimate is 5 fixed, 4 new ones ... so overall net positive :)

**Comment by sandlerr** [comment]
not sure how CC works, :+1: if this isn't going to make every subsequent PR red as well

**Comment by grosser** [comment]
cc only shows warnings / we got a ton of these ... just trying to add less then I remoe :D

**Pull request merged**